### ISSUE TITLE
feature gate logs.jsonParserArray doesn't work.

### ISSUE THREAD

**Issue by h0cheung** [open-telemetry/opentelemetry-collector-contrib#32313]
### Component(s)

pkg/stanza

## Description

After enabling feature gate `logs.jsonParserArray`, when I try to use `json_array_parser`, it still reports `unsupported type 'json_array_parser'`

## Steps to Reproduce

Try to use `json_array_parser` in a config file. Then launch otelcol-contrib with `--feature-gates=logs.jsonParserArray`.

## Expected Result

It should launch normally.

## Actual Result

It reports `unsupported type 'json_array_parser'` and exits.

### Collector version

v0.95.0

## Environment
OS: (e.g., "Ubuntu 20.04")
Compiler(if manually compiled): (e.g., "go 14.2")

### OpenTelemetry Collector configuration

```yaml
receivers: 
  filelog:
    include:
    - /tmp/test.log
    include_file_path: true
    include_file_name: false
    preserve_leading_whitespaces: true
    max_log_size: 40kb
    operators:
    - type: json_array_parser
      parse_to: body

exporters:
  debug:

service:
  pipelines:
    logs:
      receivers:
      - filelog
      exporters:
      - debug
```

### Log output

```shell
Error: failed to get config: cannot unmarshal the configuration: 1 error(s) decoding:

* error decoding 'receivers': error reading configuration for "filelog": 1 error(s) decoding:

* error decoding 'operators[0]': unsupported type 'json_array_parser'
2024/04/11 14:34:58 collector server run finished with error: failed to get config: cannot unmarshal the configuration: 1 error(s) decoding:

* error decoding 'receivers': error reading configuration for "filelog": 1 error(s) decoding:

* error decoding 'operators[0]': unsupported type 'json_array_parser'
```

### Additional context

The parser checks feature gates in `init()`, which seems to run before the command line arguements have been loaded?

**Comment by crobert-1** [comment]
_Removing `needs triage` based on code owner's response on PR that will resolve this issue._

**Comment by crobert-1** [comment]
It looks like this was resolved by https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32501, but please let me know if I misunderstood.

**Comment by li-zeyuan** [comment]
> It looks like this was resolved by #32501, but please let me know if I misunderstood.

Yes, it has been fixed.
